### ISSUE TITLE
Refine drush logging implementations

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#7918]
See my comment at https://github.com/uiowa/uiowa/pull/7901#issuecomment-2225560069

**tl;dr**

The `LoggerChannelTrait` class contains this description:
```
This utility trait should only be used in application-level code, such as
classes that would implement ContainerInjectionInterface. Services registered
in the Container should not use this trait but inject the appropriate service
directly for easier testing.
```

My read of this is that our classes should probably be calling the service or dependency injecting it rather than using the trait. The good news is that it seemed like we are actually doing both in some cases when I reviewed the work in uiowa/uiowa#5889 (to better understand how we should be handling things in uiowa/uiowa#7901), so refactoring should be pretty easy.

The usage of both the trait and the injected dependency creates potential future confusion about how the logging is actually happening and may make troubleshooting future issues more difficult.
